DNSng utils cont'd

The-Loeki opened the pull request:
### What does this PR do?
Continue adding stuff to the new DNS client stuff

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/pull/39639

### Tests written?
learning 

@cachedout I'm still here, glad you like my code so much, please add this stuff as well so I can deservedly drop the RFC/WIP/nomerge labels ;)
Also note the `service()` function which should now do SRV lookups the best possible way.

hdimitriou committed:
enabled pillarenv for state module 'state', now you can define pillarenv or if you dont state will try to find it from opts['pillarenv'] if passed on from orchestrate

hdimitriou committed:
git_pillar tries to respect pillarenv and only check when pillarenv = env, if pillarenv is set

hdimitriou committed:
fixed 'Template not found' error by adding a control in get_tops() not to look for top file if pillarenv is defined and is not equal to file_roots

hdimitriou committed:
fixed show_pillar to support pillarenv

hdimitriou committed:
added 'versionadded' on documentation change

mirceaulinic committed:
New getters: config and optics

The-Loeki committed:
DNS2: Services parser

The-Loeki committed:
DNS2: CAA record parsing, lookup tree walking, option handling

The-Loeki committed:
DNS2 Cleanup/bugfixes, SRV service()

salt-jenkins commented:
@The-Loeki, thanks for your PR! By analyzing the history of the files in this pull request, we identified @silenius, @terminalmage and @cachedout to be potential reviewers.

The-Loeki commented:
So I'm about done with the functional parts of the DNS client for now, I'm quite happy the way things are going.

Next up I'll start refactoring the existing lookup and client functions to start utilizing this code.
And of course test-test-test-test.

One question: can I rely on `__opts__` being available in `utils`? Or how do I get the running configuration?

cachedout commented:
I like the direction this going. I think we really want some unit testing in here though.

s0undt3ch committed:
Transplant configurations right after CLI parsing is complete

s0undt3ch committed:
Py3 compat

The-Loeki committed:
DNS2: Cleanup & fixes

The-Loeki committed:
DNS2: Add first batch of unit tests

s0undt3ch commented on tests/unit/utils/test_dns.py:
```diff
@@ -0,0 +1,142 @@
+# Debug
+import pprint
+ppr = pprint.PrettyPrinter(indent=2).pprint
+
```
The above pretty printer is no longer required right?

The-Loeki replied:
correct, it will disappear soon

s0undt3ch commented on tests/unit/utils/test_dns.py:
```diff
@@ -0,0 +1,142 @@
+
+# Testing
+from tests.support.unit import skipIf, TestCase
+from tests.support.mock import NO_MOCK, NO_MOCK_REASON, MagicMock, patch, call
```
If there's no mocking going on, these imports should be removed.

s0undt3ch committed:
Fix pytest loading if a system wide salt package is installed

s0undt3ch committed:
Reliable sorting on both python 2 and 3

s0undt3ch committed:
Rename decorators in order not to be picked up by pytest as test cases

s0undt3ch committed:
Update tests to the new sorting

s0undt3ch committed:
Code style fixes

s0undt3ch committed:
Skip tests on required missing libs

s0undt3ch committed:
This is not an integration test. Skip test.

Skipping tests since I have no clue what is supposed to be tested here
with that `raise Exception(ret)` at the end.

s0undt3ch committed:
Don't run these tests under Py3. The PSF takes care of that.

s0undt3ch committed:
Py3 compat

s0undt3ch committed:
Use random ports. Tests cleanup.

s0undt3ch committed:
Add file to ignored paths

s0undt3ch committed:
Take into account the comment from @damon-atkins. Improved MS Win support.

s0undt3ch committed:
Proper mocking.

s0undt3ch committed:
Pylint disable

s0undt3ch committed:
Why do need the system grains here?! This is an expensive call.

s0undt3ch committed:
Use tests.support.unit instead

s0undt3ch committed:
Minor logic fix

s0undt3ch committed:
Proper mocking

s0undt3ch committed:
This just feels wrong

defanator committed:
Fix boto_vpc.create_route() to work with interface_id

Seems like it was broken in 4611a9c5 merge commit.

s0undt3ch committed:
Style cleanup

s0undt3ch committed:
We now sync all on minion connect

s0undt3ch committed:
We need order, specially under python 3

terminalmage committed:
Preserve windows newlines in salt.template.compile_template()

Test included.

s0undt3ch committed:
Move tests to the right test module

cachedout commented:
@The-Loeki Did you get a chance to look at the above comments from @s0undt3ch ?

s0undt3ch committed:
Py3 compat

s0undt3ch committed:
Lint

s0undt3ch committed:
Take care of some more deprecated TestCase methods

s0undt3ch committed:
Stop using deprecated TestCase methods

s0undt3ch committed:
Minor lint fix

s0undt3ch committed:
Proper mocking.

s0undt3ch committed:
Error message in a single line

s0undt3ch committed:
Proper mocking.

s0undt3ch committed:
Add mixin to assert XML equality

s0undt3ch committed:
Proper mocking.

terminalmage committed:
Add userdata_renderer master config param

terminalmage committed:
Add documentation for userdata_renderer

terminalmage committed:
Add userdata_renderer fix info to 2016.11.4 release notes

terminalmage committed:
Openstack did not have templating support for userdata_file before 2016.11.4

terminalmage committed:
ec2: Add support for using userdata_renderer to template userdata_file

terminalmage committed:
Add templating support for other cloud drivers that support userdata_file

terminalmage committed:
Remove userdata_renderer value

terminalmage committed:
Rename userdata_renderer -> userdata_template in master config docs

s0undt3ch committed:
Py3 Compat

s0undt3ch committed:
Proper mocking.

terminalmage committed:
Update cloud docs to reflect userdata_renderer -> userdata_template

terminalmage committed:
Only template the userdata_file if explicitly configured to do so

terminalmage committed:
Try to read compiled template as StringIO

The fact that PyYAML was the last renderer in the render pipe initially,
masked the fact that template renderers return StringIO instances and
not strings. This commit makes sure that we try reading from the
StringIO first and then fallback to just trying to base64 encode the
userdata value as returned by compile_template().

terminalmage committed:
Update 2016.11.4 release notes for userdata_renderer -> userdata_template

s0undt3ch committed:
Py3 compat. Cast to list which works in Py2 and Py3.

s0undt3ch committed:
Proper mocking.

s0undt3ch committed:
Misplaced test. Use tests.support.unit instead.

terminalmage committed:
Move userdata templating to salt.utils.cloud

terminalmage committed:
Bring in salt.utils.stringio from develop branch

terminalmage committed:
Properly handle renderers which return StringIO objects

terminalmage committed:
Update compile_template test to use StringIO

terminalmage committed:
Allow for userdata_template to be disabled in a cloud_profile

This handles the case where a global value has been set and it is
necessary to disable it for a single profile.

techhat committed:
Add __func_alias__ back in

terminalmage committed:
Nova and openstack don't accept base64-encoded userdata

This fixes my bad copypasta.

s0undt3ch committed:
Use a temporary configuration

s0undt3ch committed:
This argument is not required when used as a ctx manager

s0undt3ch committed:
Allow creating a temporary configuration to be used for a single test

s0undt3ch committed:
Use a temporary configuration

s0undt3ch committed:
Integration tests should not use the temporary config

mirceaulinic committed:
Doc: SLS example

s0undt3ch committed:
Use a temporary configuration instead

s0undt3ch committed:
Use a temporary configuration instead

s0undt3ch committed:
Use a temporary configuration instead

mirceaulinic committed:
Fix tpyo

s0undt3ch committed:
Use a temporary configuration instead

s0undt3ch committed:
Use a temporary configuration instead

s0undt3ch committed:
Remove unused imports

mirceaulinic committed:
Pillar example & fix SLS example

mirceaulinic committed:
Netyang autodoc

mirceaulinic committed:
Index netyang state

Commit:
Fixed lint issues. Enhanced to transprently use hdfs if available and gracefully fallback on hadoop command

s0undt3ch committed:
Ignore the unimplemented abstract method

mirceaulinic committed:
Solve imports

mirceaulinic committed:
Add configured function

mirceaulinic committed:
Remove _kwargs from data

mirceaulinic committed:
Adapt new functions to Nitrogen

mirceaulinic committed:
Insert new funs into Nitrogen release notes

The-Loeki committed:
Cache roster: Rework env support through SDB

s0undt3ch committed:
Move mixins to tests.support.mixins.

The adapted config mixin get config methods are not static methods

s0undt3ch committed:
Minor import fixes

The-Loeki committed:
Cache roster lint fix

s0undt3ch committed:
Some more RUNTIME_CONFIGS and RUNTIME_VARS fixes

s0undt3ch committed:
Import for the original module

s0undt3ch committed:
Use a temporary config

s0undt3ch committed:
No need to load all modules docs.

This should reduce memory usage as well as speeding up the decorator

terminalmage committed:
Improved Docker auth handling and other misc. Docker improvements

This commit changes how we handle Docker authentication. We no longer
try to auth when pushing/pulling images. This was initially done based
on a misunderstanding of how authentication was handled in docker-py.
docker-py will automagically use the base64-encoded username/password
combination stored in config.json if present, so we don't need to auth
at all for push or pull, docker-py will handle all of that. We just need
to make sure that we get the auth info into the config.json. To do this
I have added a login() function to the execution module, which uses the
Docker CLI to authenticate (with output_loglevel set to "quiet" to
suppress any logging of the credentials). The CLI is used in this
instance because docker-py provides no interface by which to persist a
login in the config.json; it can read from the file, but is not designed
to *write* to it. Rather than trying to write to this file ourselves,
and potentially breaking it when (not *if*) Docker decides to change the
internal format of the JSON data structure, using the CLI is a way of
future-proofing the authentication logic.

Context caching of the docker-py client instance has also been removed.
Context caching was used based on the same incorrect understanding of
how docker-py handles authentication, and sought to avoid repeated login
attempts. As that is no longer a concern, there is no need to cache the
client instance in the context dunder because we don't really gain a
performance benefit from it.

The _image_wrapper() function has been removed, as it no longer serves
any purpose, and the image-specific code in it that was still needed
has been absorbed into the _client_wrapper() helper. The functions which
used the _image_wrapper() helper (like push() and pull()) now use
_client_wrapper().

Additionally, the decorators used to enforce a minimum Docker version
(or Docker API version) have been removed. These are not necessary since
docker-py handles raising exceptions when a given feature is not
supported by the effective API version. The _client_wrapper() helper
function now checks for miscellaneous docker exceptions by catching the
docker.errors.DockerException class, which is the base class for the
custom exceptions raised by docker-py, and by doing so catches exceptions
raised due to API version incompatibility (and more).

The list of functions in the top-level docstring has been removed as it
is very out-of-date, and is somewhat superfluous anyway given that we
have for some time now had a list of the functions on the right side of
the page in the documentation.

Other changes:

- Fixed a bug I introduced when I overhauled the
  docker_container.running state, in which a container with no changes
  to be made, which was not running, would be started by the state even
  when test=True was passed.
- Custom registry image names are now properly identified. The colon in
  the hostname of the custom registry was previously causing incorrect
  identification of the image name and tag, when no explicit tag was
  being passed (e.g. localhost:5000/myimage).
- When configuring credentials, the creds for the Docker Hub can be
  configured under a registry named ``hub``. This keeps the user from
  having to figure out the registry URL and configure it in their Pillar
  data, and thus makes using this module easier.
- Removes the email address from the documentation for credential
  configuration. This was probably initially added because docker-py
  accepts it as an argument, but it is entirely ignored for purposes of
  authentication (even by docker-py) and is thus unnecessary. Relic of
  an earlier time in Docker's history, perhaps?
- Fixed RST references to functions which weren't caught when this
  module was renamed to dockermod.py
- Allow filter arguments in docker.networks to be passed as a
  comma-separated list as well as a Python list, in keeping with general
  usage elsewhere in Salt.

Commit:
Merge pull request #40470 from techhat/cachetest

Add cache.store to the runners

Commit:
Fixed error message and stale docblocks

Commit:
Merge branch 'develop' into enhance/module/hadoop

The-Loeki committed:
Roster cache: Link doc to pillar.get

Whissi committed:
salt-minion.service: Start after salt-master.service

On systems where you have both services, salt-master and salt-minion,
enabled, you want that salt-minion.service starts after
salt-master.service so that the minion will be able to connect to the
master on first attempt.

s0undt3ch committed:
Move base test case modules to `tests.support.case`

q1x committed:
Added zabbix.mediatype_get and zabbix.mediatype_delete functions.

q1x committed:
Added zabbix.mediatype_create

q1x committed:
Added zabbix.mediatype_update

Mapel88 committed:
fix "set_webapp_settings"

Removes 'r' prefix from this section cause the command is not working with it...
 if setting == "userName" or setting == "password":
            pscmd.append(r" Set-WebConfigurationProperty -Filter \"system.applicationHost/sites/site[@name='{0}']/application[@path='/{1}']/virtualDirectory[@path='/']\"".format(site, name))
            pscmd.append(r" -Name \"{0}\" -Value {1};".format(setting, value))

        if setting == "physicalPath" or setting == "applicationPool":
            pscmd.append(r" Set-ItemProperty \"IIS:\Sites\{0}\{1}\" -Name {2} -Value {3};".format(site, name, setting, value))

Commit:
Merge pull request #40473 from cloudflare/napalm-yang

New execution module: Napalm YANG

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
Documentation update and encoding fix

s0undt3ch committed:
Keep the same exception raised before the move

s0undt3ch committed:
Import from the original modules not `tests.integration`

s0undt3ch committed:
Proper mocking

Commit:
Merge branch '2016.11' into 'develop'

Conflicts:
  - salt/cache/__init__.py
  - salt/grains/core.py
  - salt/utils/gitfs.py
  - tests/unit/cache/test_localfs.py
  - tests/unit/modules/test_ssh.py

Commit:
Merge branch '2016.3' into '2016.11'

Conflicts:
  - doc/topics/master_tops/index.rst
  - salt/runners/saltutil.py

Commit:
Memcache documentation and minor updates.

Commit:
Add cachedir variable setting back in after bad merge

s0undt3ch committed:
PY3 compat

Commit:
Merge pull request #40495 from rallytime/merge-2016.11

[2016.11] Merge forward from 2016.3 to 2016.11

mirceaulinic committed:
Merge branch 'develop' into add-new-funs

thatch45 committed:
Fix up the doc for failover clarity

techhat committed:
Swap around aliases

Commit:
Merge pull request #40284 from The-Loeki/cache-roster2

cache roster: Add SDB support, improve documentation

thatch45 committed:
first pass at adding support for pycryptodome installed as

pycryptodomex, this allows us to maintain compat with pycrypto being
installed on the system without introducing a module library dependency

Commit:
Group checks for failhard setting in () in state.check_failhard function

Fixes #38683

When two states combine the `require`, `failhard`, and `order` options,
the `order` option should be ignored because `require` is present. Then
the `failhard` option should make the state run fail.

The check for "failhard" in the `check_failhard` function in the state
compiler was too broad. We want "failhard" to be true AND tag to be in
the `running` dict. Without the parens around the first OR statement,
we were bypassing the AND requirement because failhard was found.

Since the second state's tag was not found in the running dict, the state
run stacktraces on a KeyError.

thatch45 committed:
fix do to pre correct on python randome function

thatch45 committed:
good catch

thatch45 committed:
fix the cryptodome version lookup for the versions report

Commit:
Merge pull request #39874 from kiorky/fixup_fail_rc

Make returncode friendly to onfail requisites

Commit:
Merge pull request #39921 from ScoreUnder/39915_fix_relpath_handling

Fix relpath handling in file.recurse

thatch45 committed:
pycryptodome adds RSA to the key header which the openssl

PEM decoder does not recognise

Commit:
Merge pull request #40102 from sitepoint/develop-file_replace_output_fix

Fix for bug in file.replace change report

Commit:
Merge pull request #40452 from q1x/develop

Added the ability to manage Zabbix usergroup permissions

Commit:
Merge pull request #40458 from cloudflare/napalm-syslog

New engine: Napalm syslog

Commit:
Merge pull request #40475 from cloudflare/yang-state

New state module: NAPALM YANG

Commit:
Merge pull request #40478 from cloudflare/add-new-funs

Add new functions to the net module

Commit:
Merge pull request #40480 from terminalmage/docker-auth-handling

Improved Docker auth handling and other misc. Docker improvements

thatch45 committed:
Add winrm framework for salt-ssh

Commit:
Merge pull request #40487 from Whissi/start-minion-after-master

salt-minion.service: Start after salt-master.service

thatch45 committed:
some tidying up

twangboy committed:
Add deprecation warning to win_update module

twangboy committed:
Add deprecation warning to win_update state module

tonybaloney committed:
Update test_libcloud_dns.py

add absolute imports

twangboy committed:
Gate deprecation warning

s0undt3ch committed:
First resource leakage fix pass

eradman committed:
Add mocks to allow keyboard tests to function on non-Linux systems

terminalmage committed:
Add initial developer docs on the fileserver and client

This may be expanded, but this gives a brief overview.

terminalmage committed:
Fix inaccurate comment

terminalmage committed:
Merge branch 'thatch45-rand_m_doc' into 2016.11

terminalmage committed:
Fix label for RST link

q1x committed:
Merge branch 'develop' of github.com:q1x/salt into develop

q1x committed:
Corrected linting error (whitespace)

The-Loeki committed:
DNS2: Add tests for getaddrinfo()

The-Loeki committed:
DNS2: Add tests for getaddrinfo()

s0undt3ch committed:
Make sure we close the spooled temporary file

Mapel88 committed:
Fixed standardization issues in win_iis

- Fixed lint error in set_webapp_settings
- Added SaltInvocationError to data input checks
- Fixed some standardization in set_webapp_settings & get_webapp_settings

s0undt3ch committed:
Make sure files are closed

s0undt3ch committed:
Use mock to patch

s0undt3ch committed:
Import from the original modules

s0undt3ch committed:
Take care of resource leakage on tests

s0undt3ch committed:
Fix mocked test

s0undt3ch committed:
Fix state files temporary path

Commit:
Set True instead of 1

s0undt3ch committed:
Use the test run temporary path, not the system temporary path

Commit:
salt/utils/aws.py: Make sure the right headers are selected for signing, and also make sure they are sorted properly.

Commit:
Merge pull request #39109 from bdrung/fix-merge-order-warning

Fix top_file_merging_strategy warning if env_order is set

carlpett committed:
Salt-vault: Fix handling of None-grains

terminalmage committed:
Add a section on the cp module

thatch45 committed:
fix lint and errors

Commit:
salt/utils/aws.py: Updated sig4 to sign _all_ headers (which is not required, but suggested). Changed construction to use join instead of repeated string concatenation.

thatch45 committed:
use salt exitcodes

Commit:
salt/utils/aws.py: Moved x-amz-content-sha256 to be included in the signed headers

terminalmage committed:
Add a section on updating the fileserver

thatch45 committed:
fix lint on the lint ignores...

Commit:
salt/utils/aws.py: Changed manual dict copying to a .copy()-call

Commit:
Merge pull request #40511 from eradman/keyboard_test

Add mocks to allow keyboard tests to function on non-Linux systems

Commit:
Merge pull request #40521 from ledeuns/patch-2

Set True instead of 1

Commit:
Merge pull request #40471 from tonybaloney/libcloud_dns_kwargs

Update to libcloud_dns module to allow custom kwargs for DNS drivers

The-Loeki committed:
DNS2: dig lookup testing

The-Loeki committed:
DNS2: dig lookup testing

The-Loeki commented:
Hi guys, I did, but those lines might be around for just a little while more while I try to teach myself some proper testing.

How do you guys like the testing setup so far? Am I doing OK or am I far out of the ballpark here?

Commit:
Remove extra blank line for lint

Commit:
Merge pull request #40339 from s0undt3ch/features/py3-pt2

[WIP] Move unit tests which were in integration tests - More py3 and mocking fixes

Commit:
Merge pull request #40468 from techhat/cachealias

Add __func_alias__ back in

Commit:
Merge branch '2016.11' into 'develop'

Conflicts:
  - salt/runners/saltutil.py

Commit:
Merge pull request #40496 from rallytime/bp-40415

Back-port #40415 to 2016.11

Commit:
[Lint] Fix spaces around variable assignment

Commit:
Added support on AIX for disk.iostat

s0undt3ch committed:
Use the more portable helper `skip_if_not_root`.

s0undt3ch committed:
Make sure the socket's used are closed

Commit:
Allow for nightly build designations in Salt versions

Commit:
Merge pull request #40525 from dmurphy18/aix_dskiostat

Add support for disk.iostat on AIX

Commit:
Merge pull request #40522 from carlpett/vault-fix-master-minion

Salt-vault: Fix handling of None-grains

Commit:
Merge pull request #40516 from q1x/develop

Mediatype support for Zabbix execution module

Commit:
Merge pull request #40513 from terminalmage/fileserver-design-docs

Add initial developer docs on the fileserver and client

Commit:
Merge pull request #40509 from thatch45/saltwinshell

Framework in place for salt-ssh to windows

Commit:
Merge pull request #40503 from thatch45/2016.11

first pass at adding support for pycryptodome installed as

Commit:
Update release information for new AIX support

Commit:
Merge pull request #40504 from rallytime/fix-38683

Group checks for failhard setting in () in state.check_failhard function

Commit:
Merge pull request #40497 from DSRCorporation/features/39275_memcache

Memcache documentation and minor updates.

Commit:
Merge pull request #40493 from Mapel88/develop

fix "set_webapp_settings"

Commit:
Added further support for functionality on AIX for 2016.11.4

Commit:
Merge pull request #40465 from rallytime/fix-37699

Artifactory Execution & State Module: Fixup Error Handling

Commit:
Remove dead code

Commit:
Merge pull request #40423 from akelge/develop

Quick patch to support SAN addition/removal

Commit:
Merge pull request #40528 from dmurphy18/salt_nightlybuild

Allow for nightly build designations in Salt versions

cachedout commented:
@The-Loeki I took at look at the tests here and they look great. Please let me know when you think this is ready for final review. Thanks!

Commit:
reverted back to split vs splitlines.

s0undt3ch committed:
Let's use the system temp dir as it was to see if it fixes the test failure

s0undt3ch committed:
Purge the account

s0undt3ch committed:
Fix state logic

tonybaloney committed:
Update hacking.rst

Small error in the instructions

tonybaloney committed:
Update hacking.rst

this one is wrong too

twangboy committed:
Move deprecation to Flourine

twangboy committed:
Fix unit tests, spell Fluorine correctly

Commit:
salt/utils/aws.py: Removed salt.ext.six import as the code that used it got removed.

s0undt3ch committed:
Merge pull request #40535 from tonybaloney/doc_fix

Update hacking.rst

q1x committed:
corrected linting errors

s0undt3ch committed:
Get some more insights on memory consumption

Commit:
A quick fix for Cache has no 'list' attribute.

bbinet committed:
Update PillarStack stack.py to latest upstream version

which provides a fix for Windows minions:
https://github.com/bbinet/pillarstack/pull/29

and the following enhancements:
https://github.com/bbinet/pillarstack/pull/26
https://github.com/bbinet/pillarstack/pull/27

s0undt3ch committed:
Limit the size of the log records queue

Commit:
Merge pull request #40530 from dmurphy18/aix_docupd

Update release information for 2016.11.4 for additional AIX support

Commit:
Fix a couple more Oxygen --> Fluorine references

Commit:
Merge pull request #40523 from s0undt3ch/features/py3

[WIP] A Little House Keeping and Addressing Py3 test suite leakage

Commit:
Merge pull request #40538 from q1x/develop

Support for Zabbix mediatypes in states

Commit:
Merge pull request #40536 from github-abcde/utils-aws-header-fix

Utils aws sig4 header fix

Commit:
Merge pull request #40472 from daxroc/enhance/module/hadoop

Enhance/module/hadoop

Commit:
Merge pull request #40464 from terminalmage/userdata-renderer

salt-cloud: Do not pass userdata_file through yaml renderer

Commit:
Merge pull request #40510 from twangboy/deprecate_win_update

Add deprecation warnings to win_update state and execution modules

Commit:
Merge pull request #40540 from DSRCorporation/bugs/40494_merge_forward_cache_list_fix

A quick fix for Cache has no 'list' attribute.

Commit:
Merge branch '2016.11' into 'develop'

Conflicts:
  - salt/template.py
  - salt/utils/pycrypto.py
  - tests/integration/modules/test_state.py
  - tests/unit/states/test_artifactory.py
  - tests/unit/test_template.py

Commit:
Merge pull request #40494 from rallytime/merge-develop

[develop] Merge forward from 2016.11 to develop

Commit:
Merge pull request #40541 from bbinet/upstream-pillarstack

Update PillarStack stack.py to latest upstream version

Commit:
Merge pull request #40531 from terminalmage/pr-35740

Rebase/resubmit of PR #35740

cachedout commented:
@The-Loeki This will also need a rebase because there are now merge conflicts. Thanks.

s0undt3ch committed:
Add missing import

s0undt3ch committed:
Add missing futures import

s0undt3ch committed:
Remove unmocked patching

s0undt3ch committed:
Fix resource leakage

s0undt3ch committed:
White-space lint fix

s0undt3ch committed:
Don't instantiate the log handler if unable to connect to server

s0undt3ch committed:
Py3 compat

s0undt3ch committed:
Merge pull request #40550 from s0undt3ch/features/py3-pt2

Resource leakage fixes

The-Loeki committed:
DNS2: bugfixes, cleanup, drill tests

The-Loeki committed:
Merge branch 'dns-enh' of github.com:The-Loeki/salt into dns-enh

The-Loeki commented:
crap... that borked it up again :(
